### PR TITLE
Added hapi BF16 lenet script

### PR DESCRIPTION
**Pull request by arlesniak**
### PR types

Others
### PR changes

Others
### Describe

Added hapi BF16 lenet script

### CODE REVIEW

**Review by jczaja — DISMISSED** [review]
LGTM

File: python/paddle/tests/hapi_mnist_bf16_static.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    labels = [Input([None, 1], 'int64', 'label')]
+
+    model = Model(LeNet(), inputs, labels)
+    optim = paddle.optimizer.SGD(learning_rate=0.001)
+    if args.bf16:
+        optim = amp.bf16.decorate_bf16(
+            optim,
+            amp_lists=amp.bf16.AutoMixedPrecisionListsBF16(
+                custom_bf16_list={
+                    'matmul_v2', 'pool2d', 'relu', 'scale', 'elementwise_add',
```
Comment (wozna) [inline python/paddle/tests/hapi_mnist_bf16_static.py:97]: Out of curiosity, are all these operators in this model and computed in BF16?

Comment (arlesniak) [reply python/paddle/tests/hapi_mnist_bf16_static.py:97]: Yes they are :) the training run in general contains ops from fwd which is explicit in model, as well as implicit bkwd ops and computation of loss, regularizer, optimizer.

**Review by wozna — DISMISSED** [review]
LGTM

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

**Review by jakpiase — DISMISSED** [review]
LGTM

**Review by jakpiase — APPROVED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@Aganlengzi Please review this PR.

**Review by Aganlengzi — APPROVED** [review]
LGTM
